scripts/ci/wasm_unpack.js

```diff
@@ -0,0 +1,235 @@
+(function(source, uncompressedSize) {
+	var base64DecToArr = (Buffer !== undefined && typeof Buffer.from === "function") ?
+	function (sB64Enc) { return Buffer.from(sB64Enc, 'base64'); } : function (sB64Enc) {
+		/*\
+		|*|
+		|*|  Base64 / binary data / UTF-8 strings utilities (#1)
+		|*|
+		|*|  https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+		|*|
+		|*|  Author: madmurphy
+		|*|
+		\*/
+		/* Array of bytes to base64 string decoding */
+		function b64ToUint6 (nChr) {
+			return nChr > 64 && nChr < 91 ?
+				nChr - 65
+				: nChr > 96 && nChr < 123 ?
+				nChr - 71
+				: nChr > 47 && nChr < 58 ?
+				nChr + 4
+				: nChr === 43 ?
+				62
+				: nChr === 47 ?
+				63
+				:
+				0;
+		}
+
+		var nInLen = sB64Enc.length,
+			nOutLen = nInLen * 3 + 1 >>> 2,
+			aBytes = new Uint8Array(nOutLen);
+
+		for (var nMod3, nMod4, nUint24 = 0, nOutIdx = 0, nInIdx = 0; nInIdx < nInLen; nInIdx++) {
+			nMod4 = nInIdx & 3;
+			nUint24 |= b64ToUint6(sB64Enc.charCodeAt(nInIdx)) << 18 - 6 * nMod4;
+			if (nMod4 === 3 || nInLen - nInIdx === 1) {
+				for (nMod3 = 0; nMod3 < 3 && nOutIdx < nOutLen; nMod3++, nOutIdx++) {
+					aBytes[nOutIdx] = nUint24 >>> (16 >>> nMod3 & 24) & 255;
+				}
+				nUint24 = 0;
+			}
+		}
+		return aBytes;
+	}
+	function lz4Decompress (src, dst) {
+		// lz4.js - An implementation of Lz4 in plain JavaScript.
+		// Original source: https://github.com/Benzinga/lz4js
+
+		// Reads a 32-bit little-endian integer from an array.
+		function readU32(b, n) {
+			var x = 0;
+			x |= b[n++] << 0;
+			x |= b[n++] << 8;
+			x |= b[n++] << 16;
+			x |= b[n++] << 24;
+			return x;
+		};
+
+		// Constants
+		// --
+
+		// Compression format parameters/constants.
+		var minMatch = 4;
+
+		// Frame constants.
+		var magicNum = 0x184D2204;
+
+		// Frame descriptor flags.
+		var fdContentChksum = 0x4;
+		var fdContentSize = 0x8;
+		var fdBlockChksum = 0x10;
+		// var fdBlockIndep = 0x20;
+		var fdVersion = 0x40;
+		var fdVersionMask = 0xC0;
+
+		// Block sizes.
+		var bsUncompressed = 0x80000000;
+		var bsShift = 4;
+		var bsMask = 7;
+		var bsMap = {
+			4: 0x10000,
+			5: 0x40000,
+			6: 0x100000,
+			7: 0x400000
+		};
+
+		// Utility functions/primitives
+		// --
+
+		// Decompresses a block of Lz4.
+		function decompressBlock (src, dst, sIndex, sLength, dIndex) {
+			var mLength, mOffset, sEnd, n, i;
+			var hasCopyWithin = dst.copyWithin !== undefined && dst.fill !== undefined;
```

Comment:
Actually when trying to test this while mocking the unavailability of ``copyWithin``, it turned out emscripten relies on it being available anyways, so we can just use it directly :-).